test/box-luatest/memtx_hash_test.lua

```diff
@@ -0,0 +1,57 @@
+local server = require('luatest.server')
+local t = require('luatest')
+
+local g = t.group()
+
+g.before_all(function(cg)
+    cg.server = server:new()
+    cg.server:start()
+end)
+
+g.after_all(function(cg)
+    cg.server:drop()
+end)
+
+g.test_insert_int_to_double = function(cg)
+    cg.server:exec(function ()
+        local ffi = require('ffi')
+
+        local s = box.schema.space.create('s')
+        s:create_index('s_idx', {type='HASH', parts={1, 'double'}, unique=true})
```

Comment:
1. The name of the test file is too generic, like it contains all memtx hash tests, but I doubt we'll add any other tests to it. For bug fixes we usually create files named like `gh_XXX_brief_description_test.lua`. In this case, I think we should name it `gh_7483_insert_int_into_double_field_test.lua`.
2. The test should work just fine for the tree index and for vinyl. So I think we should parameterize it and move it to the `engine-luatest` directory.

   <details><summary>diff</summary>
   
   ```diff
   diff --git a/test/box-luatest/memtx_hash_test.lua b/test/box-luatest/memtx_hash_test.lua
   index c349205b8319..da754e933c72 100644
   --- a/test/box-luatest/memtx_hash_test.lua
   +++ b/test/box-luatest/memtx_hash_test.lua
   @@ -1,7 +1,11 @@
    local server = require('luatest.server')
    local t = require('luatest')
    
   -local g = t.group()
   +local g = t.group(nil, {
   +    {engine = 'memtx', index_type = 'hash'},
   +    {engine = 'memtx', index_type = 'tree'},
   +    {engine = 'vinyl', index_type = 'tree'},
   +})
    
    g.before_all(function(cg)
        cg.server = server:new()
   @@ -13,11 +17,11 @@ g.after_all(function(cg)
    end)
    
    g.test_insert_int_to_double = function(cg)
   -    cg.server:exec(function ()
   +    cg.server:exec(function(engine, index_type)
            local ffi = require('ffi')
    
   -        local s = box.schema.space.create('s')
   -        s:create_index('s_idx', {type='HASH', parts={1, 'double'}, unique=true})
   +        local s = box.schema.space.create('s', {engine = engine})
   +        s:create_index('s_idx', {type = index_type, parts = {1, 'double'}})
    
            -- For double-typed key we cast it to double before hashing.
            -- So 1 and 1.0 are interpret as the same value.
   @@ -53,5 +57,5 @@ g.test_insert_int_to_double = function(cg)
            t.assert_equals(s:select({double_2e63}), {{int_2e63_minus_1}})
    
            s:drop()
   -    end)
   +    end, {cg.params.engine, cg.params.index_type})
    end
   ```
   
   </details>

3. A test should remove everything it created in an `after_test` handler so that if this particular test case fails, it doesn't affect other test cases.

   <details><summary>diff</summary>
   
   ```diff
   diff --git a/test/box-luatest/memtx_hash_test.lua b/test/box-luatest/memtx_hash_test.lua
   index c349205b8319..fa4ed9f7a2c6 100644
   --- a/test/box-luatest/memtx_hash_test.lua
   +++ b/test/box-luatest/memtx_hash_test.lua
   @@ -51,7 +51,13 @@ g.test_insert_int_to_double = function(cg)
    
            -- Select of the double should retrieve the inserted integer.
            t.assert_equals(s:select({double_2e63}), {{int_2e63_minus_1}})
   -
   -        s:drop()
        end)
    end
   +
   +g.after_test('test_insert_int_to_double', function(cg)
   +    cg.server:exec(function()
   +        if box.space.s ~= nil then
   +            box.space.s:drop()
   +        end
   +    end)
   +end)
   ```
   
   </details>

Reply:
1. Done.
2. Right, done.
3. Done, thanks for the tip.